src/generators/utils/pagingOperations.ts

```diff
@@ -0,0 +1,378 @@
+      const pagingMethodSettings: PagingMethodSettings = {
+        initialMethod: {
+          name: `${operation.namePrefix}${initialOperationName}`,
+          parameters: initialMethodParameters
+        },
+        nextMethod: nextMethodParameters
+          ? {
+              name: `${operation.namePrefix}${nextOperationName}`,
+              parameters: nextMethodParameters
+            }
+          : undefined,
+        publicMethod: {
+          name: getPublicMethodName(initialOperationName),
+          parameters: initialMethodParameters
+        },
+        allMethod: {
+          name: `${initialOperationName}PagingAll`,
+          parameters: initialMethodParameters
+        },
+        pageMethod: {
+          name: `${initialOperationName}PagingPage`,
+          parameters: initialMethodParameters
+        },
+        bodyResponseType: bodyResponseTypeName,
+        nextLinkName: operation.pagination?.nextLinkName || "nextLink",
+        itemName: operation.pagination?.itemName || "value"
+      };
+
+      writePublicMethod(operation, operationGroupClass, pagingMethodSettings);
+      writePageMethod(operation, operationGroupClass, pagingMethodSettings);
+      writeAllMethod(operationGroupClass, pagingMethodSettings);
+    });
+}
+
+/**
+ * This method enforces Azure SDK Typescript guideline that paging methods should be named list*
+ * https://azure.github.io/azure-sdk/typescript_design.html#ts-pagination-provide-list
+ */
+function getPublicMethodName(initialOperationName: string) {
+  if (initialOperationName.indexOf("list") === 0) {
+    return initialOperationName;
+  }
+
+  if (initialOperationName.indexOf("get") === 0) {
+    return initialOperationName.replace("get", "list");
+  }
+
+  const initialName = normalizeName(initialOperationName, NameType.Class);
+  return `list${initialName}`;
+}
+
+/**
+ * Generates the content of the public method, here we reference the other 2 methods All and Page
+ */
+function writePublicMethod(
+  operation: OperationDetails,
+  operationGroupClass: ClassDeclaration,
+  pagingMethodSettings: PagingMethodSettings
+) {
+  const returnType = `PagedAsyncIterableIterator<${pagingMethodSettings.bodyResponseType}, ${pagingMethodSettings.bodyResponseType}[]>`;
+
+  const method = operationGroupClass.addMethod({
+    name: pagingMethodSettings.publicMethod.name,
+    parameters: pagingMethodSettings.publicMethod.parameters,
+    scope: Scope.Public,
+    returnType,
+    docs: [
+      generateOperationJSDoc(
+        pagingMethodSettings.publicMethod.parameters,
+        operation.description
+      )
+    ]
+  });
+
+  // Extract the parameter names for the All method to call it
+  let allMethodParameters = pagingMethodSettings.allMethod.parameters
+    .map(p => p.name)
+    .join(",");
+
+  // Extract the parameter names for the page method to call it
+  const pageMethodNameParams = pagingMethodSettings.pageMethod.parameters
+    .map(p => p.name)
+    .join(",");
```

Comment:
commas is the default behavior so perhaps we can be less verbose here as well unless you prefer to keep it for readability.
```suggestion
    .join();
```